checkoutmanager/config.py

```diff
@@ -116,6 +116,20 @@ def directories(self, group=None):
                 result.append(dirinfoclass(directory, url))
         return sorted(result)
 
+    def directory_from_url(self, url):
+        for dir_info in self.directories():
+            if dir_info.url == url:
+                return dir_info
+
+    def directory_from_path(self, abspath, allow_ancestors=True):
+        for dir_info in self.directories():
+            if dir_info.directory == abspath:
+                return dir_info
+        if allow_ancestors:
+            parent = os.path.split(abspath)[0]
```

Comment:
I'll change it myself after merging, but for the record: `os.path.split(something)[0]` is equivalent to `os.path.dirname(something)`. See https://docs.python.org/2/library/os.path.html#os.path.dirname

One of those little things that I picked up a few months ago :-)